service/history/queues/executable.go

```diff
@@ -398,9 +406,17 @@ func (e *executableImpl) HandleErr(err error) (retErr error) {
 		return nil
 	}
 
+	e.attemptsWithUnexpectedError++
+	if e.attemptsWithUnexpectedError >= e.attemptsBeforeSendingToDlq() && e.dlqEnabled() {
+		e.logger.Error("Marking task as terminally failed after maximum number of attempts with unexpected errors, will send to DLQ",
+			tag.Attempt(int32(e.attemptsWithUnexpectedError)), tag.Error(err))
+		e.terminalFailureCause = fmt.Errorf("%w: %w", ErrMaxAttempts, err)
```

Comment:
Can you check what happens if terminalFailureCause is set but dlq is disabled on next retry? Will the task be dropped?

Reply:
Added a test. Yes, the task will be dropped in this case.

Reply:
That's an issue needs to be fix right? It means now tasks with unknown errors (not errors like corruption that we know are safe to drop) will be dropped even when dlq is disabled.

Reply:
I have added a condition to avoid this.